tokio/src/runtime/task/core.rs

```diff
@@ -120,6 +123,7 @@ impl<T: Future, S: Schedule> Cell<T, S> {
                 stage: CoreStage {
                     stage: UnsafeCell::new(Stage::Running(future)),
                 },
+                task_id,
```

Comment:
*if* we store this, it should probably be in the trailer no?

Reply:
The header needs to stay under a cache line, but as long as CI passes, it should be good to go: https://github.com/tokio-rs/tokio/blob/master/tokio/src/runtime/task/core.rs#L266

Reply:
It has to be stored somewhere in the task structure because it's passed to `poll_future` and `cancel_task` in order to populate a `JoinError`:
https://github.com/tokio-rs/tokio/blob/a201b91a245f2558ed58f0b516af79df63e4fc77/tokio/src/runtime/task/harness.rs#L444-L451
and
https://github.com/tokio-rs/tokio/blob/a201b91a245f2558ed58f0b516af79df63e4fc77/tokio/src/runtime/task/harness.rs#L479

It could go in the `Core` or in the `Trailer` --- my  sense was that the `Core` contained data that's used when actually polling the task, and the trailer contains data that's only used when the task completes. _Technically_ the `Id` is consumed only on completion, but inside of the functions that are called on every poll, so we would need to dereference the trailer in those functions to get the ID, and we don't currently dereference the trailer on polls.

I agree that it shouldn't go in the header; it currently doesn't.